llama_stack/core/resolver.py

```diff
@@ -390,6 +390,10 @@ async def instantiate_provider(
         if "policy" in inspect.signature(getattr(module, method)).parameters:
             args.append(policy)
 
+    # vector_io providers need access to run_config.vector_stores
+    if provider_spec.api == Api.vector_io and "run_config" in inspect.signature(getattr(module, method)).parameters:
```

Comment:
hmm the router should get access to it but not the providers?

Reply:
so vector_io doesn't always go through router.py, which means we have to add models_api to the providers and have them do the check there.

if we are okay with vector_io going through the router, then we can add the default check there and pass that in the model_extra.

lmk which you'd prefer

Reply:
talked this over on Discord.